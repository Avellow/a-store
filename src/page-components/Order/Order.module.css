.order {
  display: flex;
  gap: 40px;
  justify-content: center;
}

.form {
  min-width: 440px;
}

.cart {
  max-width: 500px;
}

.items {
  list-style: none;
  padding: 0;
  margin: 0;
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.amount {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  gap: 10px;

  padding: 15px;

  position: sticky;
  bottom: 0;
  background-color: white;
  margin: 5px;
  box-shadow: 0 3.2px 7.2px 0 rgb(0 0 0 / 13%), 0 0.6px 1.8px 0 rgb(0 0 0 / 11%);
}

.price {
  font-family: var(--my-font-family);
  font-weight: 500;
}

.total {
  font-family: var(--my-font-family);
  font-weight: 700;
  font-style: italic;
}

.header {
  font-family: var(--my-font-family);
  font-style: italic;
}

.backArrow {
  cursor: pointer;
}

@media screen and (max-width: 990px) {
  .order {
    flex-direction: column-reverse;
    align-items: center;
  }

  .form {
    min-width: 300px;
  }

  .cart {
    max-width: 600px;
  }
}
