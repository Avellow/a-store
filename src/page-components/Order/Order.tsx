import { useReducer } from 'react';
import { v1 } from 'uuid';
import { Typography } from '@alfalab/core-components/typography';
import { Amount } from '@alfalab/core-components/amount';
import { ModalResponsive } from '@alfalab/core-components/modal/responsive';
import { ModalResponsiveProps } from '@alfalab/core-components/modal/typings';
import { ArrowBackMIcon } from '@alfalab/icons-glyph/ArrowBackMIcon';

import { useAppSelector } from '../../store';
import { cartItemsSelector, totalAmountSelector } from '../../store/cart';
import { CartItem, Divider, Form } from '../../components';
import styles from './Order.module.css';
import { DeliveryEnum, deliveryReducer } from './delivery.reducer';

const initialState = { name: 'Доставка по России', price: 350 }

export const Order = (): JSX.Element => {

  const cartItems = useAppSelector(cartItemsSelector);
  const totalCartPrice = useAppSelector(totalAmountSelector);

  const [delivery, dispatch] = useReducer(deliveryReducer, initialState);

  const handleDeliveryChange = (newDelivery: DeliveryEnum) => {
    dispatch({ type: newDelivery });
  };

  return (
    <div className={styles.order}>

      <Form className={styles.form} onDeliveryChange={handleDeliveryChange} data-test-id='form' />

      <div className={styles.cart} data-test-id='items'>
        <ul className={styles.items} data-test-id='cart-items-list'>
          {cartItems.map(item => (
            <li key={v1()} data-test-id='cart-item'>
              <CartItem item={item} />
            </li>
          ))}
        </ul>
        <Divider />
        <div className={styles.amount} data-test-id='total'>
          <Typography.Text tag='div' view='primary-small' className={styles.price}>
            Subtotal: <Amount value={totalCartPrice} minority={0} currency='RUB' bold='none' />
          </Typography.Text>
          <Typography.Text tag='div' view='primary-small' className={styles.price}>
            {delivery.name}: <Amount value={delivery.price} minority={0} currency='RUB' bold='none' />
          </Typography.Text>
          <Typography.Text weight='bold' tag='div' view='primary-large' className={styles.total}>
            Total: <Amount value={totalCartPrice + delivery.price} minority={0} currency='RUB' bold='full' />
          </Typography.Text>
        </div>
      </div>

    </div>
  );
};


export const OrderPanel = (
  { onAddonClick, ...props }: ModalResponsiveProps & { onAddonClick?: () => void }
): JSX.Element => {

  const handleAddonClick = () => onAddonClick && onAddonClick();

  return (
    <ModalResponsive fullscreen {...props}>
      <ModalResponsive.Header
        hasCloser
        title='Ваш заказ'
        align='center'
        leftAddons={<ArrowBackMIcon onClick={handleAddonClick} className={styles.backArrow} />}
        contentClassName={styles.header}
      />
      <ModalResponsive.Content>
        <Order />
      </ModalResponsive.Content>
    </ModalResponsive>
  );
};
