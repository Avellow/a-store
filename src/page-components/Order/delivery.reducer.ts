export enum DeliveryEnum {
  pickup = "pickup",
  moscow = "moscow",
  russia = "russia",
}

interface ChangeDeliveryAction {
  type: DeliveryEnum;
}

export interface DeliveryState {
  name: string;
  price: number;
}

export const deliveryReducer = (
  state: DeliveryState,
  { type }: ChangeDeliveryAction
) => {
  switch (type) {
    case DeliveryEnum.pickup:
      return { ...state, name: "Самовывоз", price: 0 };
    case DeliveryEnum.moscow:
      return { ...state, name: "Курьером по Москве", price: 300 };
    case DeliveryEnum.russia:
      return { ...state, name: "Доставка по России", price: 350 };
    default:
      throw new Error("Такого типа доставки нет");
  }
};
