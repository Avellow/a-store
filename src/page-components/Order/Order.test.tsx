import { configure, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { Order } from './Order';
import { renderWithRouterAndProvider } from '../../tests/helpers/renderWithRouterAndProvider';

configure({ testIdAttribute: 'data-test-id' });

describe('Order component view', () => {

  const render = () => renderWithRouterAndProvider(<Order />);

  it('should be rendered', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should render form, items, total price', () => {
    render();
    expect(screen.getByTestId('form')).toBeInTheDocument();
    expect(screen.getByTestId('items')).toBeInTheDocument();
    expect(screen.getByTestId('total')).toBeInTheDocument();
  });
});
