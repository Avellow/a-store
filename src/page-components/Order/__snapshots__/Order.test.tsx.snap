// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component view should be rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="order"
      >
        <form
          class="form form"
          data-test-id="form"
        >
          <div
            class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
          >
            <div
              class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
            >
              <div
                class="form-control__inputWrapper_3j482"
              >
                <span
                  aria-hidden="true"
                  class="form-control__hiddenLabel_3j482"
                >
                  ФИО
                </span>
                <div
                  class="form-control__label_3j482 form-control__label_tvifb label"
                >
                  <span
                    class="form-control__labelInner_3j482"
                  >
                    ФИО
                  </span>
                </div>
                <div
                  class="form-control__input_3j482"
                >
                  <input
                    aria-label="ФИО"
                    class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                    data-test-id="name-input"
                    name="name"
                    placeholder="Фамилия Имя Отчество"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
          >
            <div
              class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
            >
              <div
                class="form-control__inputWrapper_3j482"
              >
                <span
                  aria-hidden="true"
                  class="form-control__hiddenLabel_3j482"
                >
                  email
                </span>
                <div
                  class="form-control__label_3j482 form-control__label_tvifb label"
                >
                  <span
                    class="form-control__labelInner_3j482"
                  >
                    email
                  </span>
                </div>
                <div
                  class="form-control__input_3j482"
                >
                  <input
                    aria-label="email"
                    class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                    data-test-id="email-input"
                    name="email"
                    placeholder="example@site.ru"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
          >
            <div
              class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
            >
              <div
                class="form-control__inputWrapper_3j482"
              >
                <span
                  aria-hidden="true"
                  class="form-control__hiddenLabel_3j482"
                >
                  Номер телефона
                </span>
                <div
                  class="form-control__label_3j482 form-control__label_tvifb label"
                >
                  <span
                    class="form-control__labelInner_3j482"
                  >
                    Номер телефона
                  </span>
                </div>
                <div
                  class="form-control__input_3j482"
                >
                  <input
                    aria-label="Номер телефона"
                    class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                    data-test-id="phone-input"
                    name="phone"
                    placeholder="+7 000 000-00-00"
                    type="tel"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
          >
            <div
              class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
            >
              <div
                class="form-control__inputWrapper_3j482"
              >
                <span
                  aria-hidden="true"
                  class="form-control__hiddenLabel_3j482"
                >
                  Адрес (если выбран самовывоз — оставьте поле пустым)
                </span>
                <div
                  class="form-control__label_3j482 form-control__label_tvifb label"
                >
                  <span
                    class="form-control__labelInner_3j482"
                  >
                    Адрес (если выбран самовывоз — оставьте поле пустым)
                  </span>
                </div>
                <div
                  class="form-control__input_3j482"
                >
                  <input
                    aria-label="Адрес (если выбран самовывоз — оставьте поле пустым)"
                    class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                    data-test-id="address-input"
                    name="address"
                    placeholder="Индекс, город, улица, дом, квартира"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="radio-group__component_h0fij radio-group__radio_h0fij radio-group__vertical_h0fij radioGroup"
            data-test-id="delivery-radio"
          >
            <span
              class="radio-group__label_h0fij"
            >
              Доставка
            </span>
            <div
              class="radio-group__radioList_h0fij"
            >
              <label
                class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
              >
                <input
                  data-test-id="radio-Доставка по России — 350₽"
                  type="radio"
                  value="Доставка по России — 350₽"
                />
                <span
                  class="radio__circle_ajohh"
                />
                <span
                  class="radio__content_ajohh"
                >
                  <span
                    class="radio__label_ajohh"
                  >
                    Доставка по России - 350₽
                  </span>
                </span>
              </label>
              <label
                class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
              >
                <input
                  data-test-id="radio-Курьером по Москве — 300₽"
                  type="radio"
                  value="Курьером по Москве — 300₽"
                />
                <span
                  class="radio__circle_ajohh"
                />
                <span
                  class="radio__content_ajohh"
                >
                  <span
                    class="radio__label_ajohh"
                  >
                    Курьером по Москве - 300₽
                  </span>
                </span>
              </label>
              <label
                class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
              >
                <input
                  data-test-id="radio-Самовывоз (пр-т Андропова, 18 корп. 3)"
                  type="radio"
                  value="Самовывоз (пр-т Андропова, 18 корп. 3)"
                />
                <span
                  class="radio__circle_ajohh"
                />
                <span
                  class="radio__content_ajohh"
                >
                  <span
                    class="radio__label_ajohh"
                  >
                    Самовывоз (пр-т Андропова, 18 корп. 3)
                  </span>
                </span>
              </label>
            </div>
          </div>
          <div
            class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
          >
            <div
              class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
            >
              <div
                class="form-control__inputWrapper_3j482"
              >
                <span
                  aria-hidden="true"
                  class="form-control__hiddenLabel_3j482"
                >
                  Промокод
                </span>
                <div
                  class="form-control__label_3j482 form-control__label_tvifb label"
                >
                  <span
                    class="form-control__labelInner_3j482"
                  >
                    Промокод
                  </span>
                </div>
                <div
                  class="form-control__input_3j482"
                >
                  <input
                    aria-label="Промокод"
                    class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                    data-test-id="promo-input"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <label
            class="checkbox__component_1x1bs checkbox__s_1x1bs checkbox__start_1x1bs privacy"
          >
            <input
              data-test-id="privacy-check"
              type="checkbox"
            />
            <span
              class="checkbox__box_1x1bs"
            />
            <span
              class="checkbox__content_1x1bs"
            >
              <span
                class="checkbox__label_1x1bs"
              >
                Согласен с политикой конфиденциальности и обработки персональных данных
              </span>
            </span>
          </label>
          <div
            class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
          >
            <div
              class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
            >
              <div
                class="form-control__inputWrapper_3j482"
              >
                <span
                  aria-hidden="true"
                  class="form-control__hiddenLabel_3j482"
                >
                  Комментарий к заказу
                </span>
                <div
                  class="form-control__label_3j482 form-control__label_tvifb"
                >
                  <span
                    class="form-control__labelInner_3j482"
                  >
                    Комментарий к заказу
                  </span>
                </div>
                <div
                  class="form-control__input_3j482"
                >
                  <textarea
                    autocomplete="on"
                    class="textarea__textarea_1ndyb textarea__textarea_cdfdx textarea__l_1ndyb textarea__hasInnerLabel_1ndyb textarea__hasInnerLabel_cdfdx textarea__resizeVertical_1ndyb"
                    data-test-id="comment-input"
                    name="comment"
                    rows="1"
                  />
                </div>
              </div>
            </div>
          </div>
          <span
            class="paymentHint typography__primary-small_xcdcz"
          >
            Выберите способ оплаты «Промокод», если ваш заказ не превышает сумму промокода. Если больше — выберите оплату картой.
          </span>
          <div
            class="radio-group__component_h0fij radio-group__radio_h0fij radio-group__vertical_h0fij"
            data-test-id="payment-radio"
          >
            <span
              class="radio-group__label_h0fij"
            >
              Способ оплаты
            </span>
            <div
              class="radio-group__radioList_h0fij"
            >
              <label
                class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
              >
                <input
                  data-test-id="radio-Банковская карта"
                  type="radio"
                  value="Банковская карта"
                />
                <span
                  class="radio__circle_ajohh"
                />
                <span
                  class="radio__content_ajohh"
                >
                  <span
                    class="radio__label_ajohh"
                  >
                    Банковская карта
                  </span>
                </span>
              </label>
              <label
                class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
              >
                <input
                  data-test-id="radio-Промокод"
                  type="radio"
                  value="Промокод"
                />
                <span
                  class="radio__circle_ajohh"
                />
                <span
                  class="radio__content_ajohh"
                >
                  <span
                    class="radio__label_ajohh"
                  >
                    Промокод
                  </span>
                </span>
              </label>
            </div>
          </div>
          <button
            class="button__component_onnkc button__m_onnkc button__component_1qjkj button__secondary_1qjkj button__block_onnkc button"
            data-test-id="submit"
            disabled=""
            type="submit"
          >
            <span
              class="button__text_onnkc button__stretchText_onnkc"
            >
              Дальше
            </span>
          </button>
        </form>
        <div
          class="cart"
          data-test-id="items"
        >
          <ul
            class="items"
            data-test-id="cart-items-list"
          />
          <hr
            class="hr"
          />
          <div
            class="amount"
            data-test-id="total"
          >
            <div
              class="price typography__primary-small_xcdcz"
            >
              Subtotal: 
              <span
                class="amount__component_96zmx"
              >
                0
                <span
                  class=""
                >
                   ₽
                </span>
              </span>
            </div>
            <div
              class="price typography__primary-small_xcdcz"
            >
              Доставка по России
              : 
              <span
                class="amount__component_96zmx"
              >
                350
                <span
                  class=""
                >
                   ₽
                </span>
              </span>
            </div>
            <div
              class="total typography__primary-large_xcdcz typography__bold_xcdcz"
            >
              Total: 
              <span
                class="amount__component_96zmx amount__bold_96zmx"
              >
                350
                <span
                  class=""
                >
                   ₽
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="order"
    >
      <form
        class="form form"
        data-test-id="form"
      >
        <div
          class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
        >
          <div
            class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
          >
            <div
              class="form-control__inputWrapper_3j482"
            >
              <span
                aria-hidden="true"
                class="form-control__hiddenLabel_3j482"
              >
                ФИО
              </span>
              <div
                class="form-control__label_3j482 form-control__label_tvifb label"
              >
                <span
                  class="form-control__labelInner_3j482"
                >
                  ФИО
                </span>
              </div>
              <div
                class="form-control__input_3j482"
              >
                <input
                  aria-label="ФИО"
                  class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                  data-test-id="name-input"
                  name="name"
                  placeholder="Фамилия Имя Отчество"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
        >
          <div
            class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
          >
            <div
              class="form-control__inputWrapper_3j482"
            >
              <span
                aria-hidden="true"
                class="form-control__hiddenLabel_3j482"
              >
                email
              </span>
              <div
                class="form-control__label_3j482 form-control__label_tvifb label"
              >
                <span
                  class="form-control__labelInner_3j482"
                >
                  email
                </span>
              </div>
              <div
                class="form-control__input_3j482"
              >
                <input
                  aria-label="email"
                  class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                  data-test-id="email-input"
                  name="email"
                  placeholder="example@site.ru"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
        >
          <div
            class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
          >
            <div
              class="form-control__inputWrapper_3j482"
            >
              <span
                aria-hidden="true"
                class="form-control__hiddenLabel_3j482"
              >
                Номер телефона
              </span>
              <div
                class="form-control__label_3j482 form-control__label_tvifb label"
              >
                <span
                  class="form-control__labelInner_3j482"
                >
                  Номер телефона
                </span>
              </div>
              <div
                class="form-control__input_3j482"
              >
                <input
                  aria-label="Номер телефона"
                  class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                  data-test-id="phone-input"
                  name="phone"
                  placeholder="+7 000 000-00-00"
                  type="tel"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
        >
          <div
            class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
          >
            <div
              class="form-control__inputWrapper_3j482"
            >
              <span
                aria-hidden="true"
                class="form-control__hiddenLabel_3j482"
              >
                Адрес (если выбран самовывоз — оставьте поле пустым)
              </span>
              <div
                class="form-control__label_3j482 form-control__label_tvifb label"
              >
                <span
                  class="form-control__labelInner_3j482"
                >
                  Адрес (если выбран самовывоз — оставьте поле пустым)
                </span>
              </div>
              <div
                class="form-control__input_3j482"
              >
                <input
                  aria-label="Адрес (если выбран самовывоз — оставьте поле пустым)"
                  class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                  data-test-id="address-input"
                  name="address"
                  placeholder="Индекс, город, улица, дом, квартира"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="radio-group__component_h0fij radio-group__radio_h0fij radio-group__vertical_h0fij radioGroup"
          data-test-id="delivery-radio"
        >
          <span
            class="radio-group__label_h0fij"
          >
            Доставка
          </span>
          <div
            class="radio-group__radioList_h0fij"
          >
            <label
              class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
            >
              <input
                data-test-id="radio-Доставка по России — 350₽"
                type="radio"
                value="Доставка по России — 350₽"
              />
              <span
                class="radio__circle_ajohh"
              />
              <span
                class="radio__content_ajohh"
              >
                <span
                  class="radio__label_ajohh"
                >
                  Доставка по России - 350₽
                </span>
              </span>
            </label>
            <label
              class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
            >
              <input
                data-test-id="radio-Курьером по Москве — 300₽"
                type="radio"
                value="Курьером по Москве — 300₽"
              />
              <span
                class="radio__circle_ajohh"
              />
              <span
                class="radio__content_ajohh"
              >
                <span
                  class="radio__label_ajohh"
                >
                  Курьером по Москве - 300₽
                </span>
              </span>
            </label>
            <label
              class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
            >
              <input
                data-test-id="radio-Самовывоз (пр-т Андропова, 18 корп. 3)"
                type="radio"
                value="Самовывоз (пр-т Андропова, 18 корп. 3)"
              />
              <span
                class="radio__circle_ajohh"
              />
              <span
                class="radio__content_ajohh"
              >
                <span
                  class="radio__label_ajohh"
                >
                  Самовывоз (пр-т Андропова, 18 корп. 3)
                </span>
              </span>
            </label>
          </div>
        </div>
        <div
          class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
        >
          <div
            class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
          >
            <div
              class="form-control__inputWrapper_3j482"
            >
              <span
                aria-hidden="true"
                class="form-control__hiddenLabel_3j482"
              >
                Промокод
              </span>
              <div
                class="form-control__label_3j482 form-control__label_tvifb label"
              >
                <span
                  class="form-control__labelInner_3j482"
                >
                  Промокод
                </span>
              </div>
              <div
                class="form-control__input_3j482"
              >
                <input
                  aria-label="Промокод"
                  class="input__input_1j8wk input__input_n0erl input__l_1j8wk input__hasInnerLabel_1j8wk input__hasInnerLabel_n0erl"
                  data-test-id="promo-input"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <label
          class="checkbox__component_1x1bs checkbox__s_1x1bs checkbox__start_1x1bs privacy"
        >
          <input
            data-test-id="privacy-check"
            type="checkbox"
          />
          <span
            class="checkbox__box_1x1bs"
          />
          <span
            class="checkbox__content_1x1bs"
          >
            <span
              class="checkbox__label_1x1bs"
            >
              Согласен с политикой конфиденциальности и обработки персональных данных
            </span>
          </span>
        </label>
        <div
          class="form-control__component_3j482 form-control__component_tvifb form-control__l_3j482 form-control__block_3j482"
        >
          <div
            class="field form-control__inner_3j482 form-control__inner_tvifb form-control__hasInnerLabel_3j482"
          >
            <div
              class="form-control__inputWrapper_3j482"
            >
              <span
                aria-hidden="true"
                class="form-control__hiddenLabel_3j482"
              >
                Комментарий к заказу
              </span>
              <div
                class="form-control__label_3j482 form-control__label_tvifb"
              >
                <span
                  class="form-control__labelInner_3j482"
                >
                  Комментарий к заказу
                </span>
              </div>
              <div
                class="form-control__input_3j482"
              >
                <textarea
                  autocomplete="on"
                  class="textarea__textarea_1ndyb textarea__textarea_cdfdx textarea__l_1ndyb textarea__hasInnerLabel_1ndyb textarea__hasInnerLabel_cdfdx textarea__resizeVertical_1ndyb"
                  data-test-id="comment-input"
                  name="comment"
                  rows="1"
                />
              </div>
            </div>
          </div>
        </div>
        <span
          class="paymentHint typography__primary-small_xcdcz"
        >
          Выберите способ оплаты «Промокод», если ваш заказ не превышает сумму промокода. Если больше — выберите оплату картой.
        </span>
        <div
          class="radio-group__component_h0fij radio-group__radio_h0fij radio-group__vertical_h0fij"
          data-test-id="payment-radio"
        >
          <span
            class="radio-group__label_h0fij"
          >
            Способ оплаты
          </span>
          <div
            class="radio-group__radioList_h0fij"
          >
            <label
              class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
            >
              <input
                data-test-id="radio-Банковская карта"
                type="radio"
                value="Банковская карта"
              />
              <span
                class="radio__circle_ajohh"
              />
              <span
                class="radio__content_ajohh"
              >
                <span
                  class="radio__label_ajohh"
                >
                  Банковская карта
                </span>
              </span>
            </label>
            <label
              class="radio__container_ajohh radio__s_ajohh radio__start_ajohh radio-group__radio_h0fij"
            >
              <input
                data-test-id="radio-Промокод"
                type="radio"
                value="Промокод"
              />
              <span
                class="radio__circle_ajohh"
              />
              <span
                class="radio__content_ajohh"
              >
                <span
                  class="radio__label_ajohh"
                >
                  Промокод
                </span>
              </span>
            </label>
          </div>
        </div>
        <button
          class="button__component_onnkc button__m_onnkc button__component_1qjkj button__secondary_1qjkj button__block_onnkc button"
          data-test-id="submit"
          disabled=""
          type="submit"
        >
          <span
            class="button__text_onnkc button__stretchText_onnkc"
          >
            Дальше
          </span>
        </button>
      </form>
      <div
        class="cart"
        data-test-id="items"
      >
        <ul
          class="items"
          data-test-id="cart-items-list"
        />
        <hr
          class="hr"
        />
        <div
          class="amount"
          data-test-id="total"
        >
          <div
            class="price typography__primary-small_xcdcz"
          >
            Subtotal: 
            <span
              class="amount__component_96zmx"
            >
              0
              <span
                class=""
              >
                 ₽
              </span>
            </span>
          </div>
          <div
            class="price typography__primary-small_xcdcz"
          >
            Доставка по России
            : 
            <span
              class="amount__component_96zmx"
            >
              350
              <span
                class=""
              >
                 ₽
              </span>
            </span>
          </div>
          <div
            class="total typography__primary-large_xcdcz typography__bold_xcdcz"
          >
            Total: 
            <span
              class="amount__component_96zmx amount__bold_96zmx"
            >
              350
              <span
                class=""
              >
                 ₽
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
